### ISSUE TITLE
Move tests to tests/ dir

### ISSUE THREAD

**Issue by mleonhard** [florian1345/lru-mem#18]
I'm interested in using this crate in a production server.  How about adding some tests?

**Comment by florian1345** [comment]
Hello, thanks for your interest.

The CI currently runs 40 "normal" tests in the `tests` module and 51 Doc-tests, although I will admit that I have not set up any coverage measurement yet. As this is just a hobby-project for me, I cannot sink large amounts of time into it. If you already know what level of test cases would make you more comfortable or which API you want to use, I can try to work towards that goal.

Of course, you are always free to PR some tests, if you like.

**Comment by mleonhard** [comment]
I searched a little harder and found the tests in `src/tests.rs`.  How about moving them to `tests/test.rs` to follow the recommended layout in [The Cargo Book - Package Layout](https://doc.rust-lang.org/cargo/guide/project-layout.html)?

You replied while I was writing this comment.

**Comment by florian1345** [comment]
Unfortunately, I cannot move all tests to the `tests` directory, as that is used for integration tests, which can only use the public API of the library, while some of my tests do assertions on internal state. I shuffled some tests around and identified two which I would consider "integration tests". They were moved as you suggested, the rest was merged into the respective module files as proposed in the guidelines [here](https://doc.rust-lang.org/book/ch11-03-test-organization.html). See #19
